genesyscloud/supported_content/resource_genesyscloud_supported_content_utils.go

```diff
@@ -0,0 +1,115 @@
+
+	for _, mediaType := range mediaTypes {
+		mediaTypesMap, ok := mediaType.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		resourcedata.BuildSDKInterfaceArrayValueIfNotNil(&sdkMediaType.Allow, mediaTypesMap, "allow", buildMediaTypeAccesss)
+	}
+	return &sdkMediaType
+}
+
+// buildMediaTypeAccesss maps an []interface{} into a Genesys Cloud *[]platformclientv2.Mediatypeaccess
+func buildMediaTypeAccesss(mediaTypeAccesss []interface{}) *platformclientv2.Mediatypeaccess {
+	var sdkMediaTypeAccess platformclientv2.Mediatypeaccess
+	for _, mediaTypeAccess := range mediaTypeAccesss {
+		mediaTypeAccesssMap, ok := mediaTypeAccess.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		resourcedata.BuildSDKInterfaceArrayValueIfNotNil(&sdkMediaTypeAccess.Inbound, mediaTypeAccesssMap, "inbound", buildMediaTypess)
+		resourcedata.BuildSDKInterfaceArrayValueIfNotNil(&sdkMediaTypeAccess.Outbound, mediaTypeAccesssMap, "outbound", buildMediaTypess)
+
+	}
+
+	return &sdkMediaTypeAccess
+}
+
+// buildMediaTypess maps an []interface{} into a Genesys Cloud *[]platformclientv2.Mediatypes
+func buildMediaTypess(mediaTypess []interface{}) *[]platformclientv2.Mediatype {
+	mediaTypessSlice := make([]platformclientv2.Mediatype, 0)
+	for _, mediaTypes := range mediaTypess {
+		var sdkMediaTypes platformclientv2.Mediatype
+		mediaTypessMap, ok := mediaTypes.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		resourcedata.BuildSDKStringValueIfNotNil(&sdkMediaTypes.VarType, mediaTypessMap, "type")
+
+		mediaTypessSlice = append(mediaTypessSlice, sdkMediaTypes)
+	}
+
+	return &mediaTypessSlice
+}
+
+// flattenMediaTypes maps a Genesys Cloud *[]platformclientv2.Mediatype into a []interface{}
+func flattenMediaTypes(mediaTypes *[]platformclientv2.Mediatype) []interface{} {
+	if len(*mediaTypes) == 0 {
+		return nil
+	}
+
+	var mediaTypeList []interface{}
+	for _, mediaType := range *mediaTypes {
+		mediaTypeMap := make(map[string]interface{})
+
+		resourcedata.SetMapValueIfNotNil(mediaTypeMap, "type", mediaType.VarType)
+
+		mediaTypeList = append(mediaTypeList, mediaTypeMap)
+	}
+
+	return mediaTypeList
+}
+
+// flattenMediaTypeAccesss maps a Genesys Cloud *[]platformclientv2.Mediatypeaccess into a []interface{}
+func flattenMediaTypeAccesss(mediaTypeAccesss *platformclientv2.Mediatypeaccess) []interface{} {
+
+	var mediaTypeAccessList []interface{}
+	mediaTypeAccessMap := make(map[string]interface{})
+
+	resourcedata.SetMapInterfaceArrayWithFuncIfNotNil(mediaTypeAccessMap, "inbound", mediaTypeAccesss.Inbound, flattenMediaTypes)
+	resourcedata.SetMapInterfaceArrayWithFuncIfNotNil(mediaTypeAccessMap, "outbound", mediaTypeAccesss.Outbound, flattenMediaTypes)
+
+	mediaTypeAccessList = append(mediaTypeAccessList, mediaTypeAccessMap)
+
+	return mediaTypeAccessList
+}
+
+// flattenMediaTypess maps a Genesys Cloud *[]platformclientv2.Mediatypes into a []interface{}
+func flattenMediaTypess(mediaTypess *platformclientv2.Mediatypes) []interface{} {
```

Comment:
There is a function called `flattenMediaTypess` and another called `flattenMediaTypes`. 

Also, if you define an inbound item like this: `inbound{}` , leaving out the `type` field, you see this error: `panic: interface conversion: interface {} is nil, not map[string]interface {}`. This is because somewhere we're casting nil to a map. It's hard to tell where this is happening looking through this file though. I think the resource generator wrote this file in a way that makes it hard to read/understand. I think it's probably a good idea to just write these functions yourself so we know what the code is doing.